pkg/provider/aws/aws.go

```diff
@@ -0,0 +1,162 @@
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/ssm"
+
+	"github.com/bank-vaults/secret-init/pkg/provider"
+)
+
+var ProviderName = "aws"
+
+type Provider struct {
+	sm  *secretsmanager.SecretsManager
+	ssm *ssm.SSM
+}
+
+func NewProvider(config *Config) *Provider {
+	return &Provider{
+		sm:  secretsmanager.New(config.session),
+		ssm: ssm.New(config.session),
+	}
+}
+
+func (p *Provider) LoadSecrets(ctx context.Context, paths []string) ([]provider.Secret, error) {
+	var secrets []provider.Secret
+
+	for _, path := range paths {
+		split := strings.SplitN(path, "=", 2)
+		originalKey, secretID := split[0], split[1]
+
+		// valid secretsmanager secret examples:
+		// arn:aws:secretsmanager:region:account-id:secret:secret-name
+		// secretsmanager:secret-name
+		if strings.Contains(secretID, "secretsmanager:") {
+			secret, err := p.sm.GetSecretValueWithContext(
+				ctx,
+				&secretsmanager.GetSecretValueInput{
+					SecretId: aws.String(secretID),
+				})
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get secret from AWS secrets manager")
+			}
+
+			secretBytes, err := extractSecretValueFromSM(secret)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to extract secret value from AWS secrets manager")
+			}
+
+			secretValue, err := parseSecretValueFromSM(secretBytes)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to parse secret value from AWS secrets manager")
+			}
+
+			switch value := secretValue.(type) {
+			case string:
+				secrets = append(secrets, provider.Secret{
+					Key:   originalKey,
+					Value: value,
+				})
+
+			case []byte:
+				secrets = append(secrets, provider.Secret{
+					Key:   originalKey,
+					Value: string(value),
+				})
+			}
+		}
+
+		// Valid ssm parameter examples:
+		// arn:aws:ssm:region:account-id:parameter/path/to/parameter-name
+		// arn:aws:ssm:us-west-2:123456789012:parameter/my-parameter
+		if strings.Contains(secretID, "ssm:") {
+			parameteredSecret, err := p.ssm.GetParameterWithContext(
+				ctx,
+				&ssm.GetParameterInput{
+					Name:           aws.String(secretID),
+					WithDecryption: aws.Bool(true),
+				})
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get parameter from AWS SSM")
+			}
+
+			secrets = append(secrets, provider.Secret{
+				Key:   originalKey,
+				Value: aws.StringValue(parameteredSecret.Parameter.Value),
+			})
+		}
+	}
+
+	return secrets, nil
+}
+
+// https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html
+func extractSecretValueFromSM(secret *secretsmanager.GetSecretValueOutput) ([]byte, error) {
+	// Secret available as string
+	if secret.SecretString != nil {
+		return []byte(aws.StringValue(secret.SecretString)), nil
+	}
+
+	// Secret available as binary
+	decodedSecret, err := base64.StdEncoding.DecodeString(string(secret.SecretBinary))
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode secret: %w", err)
+	}
```

Comment:
the AWS Go SDK docs indicate that the value of this property is available decoded already, but I didnt fully verify this. the previous fallback _invalid base64 bytes error_ was there to allow us to try to decode if the binary data is valid base64 string or simply return the contents when invalid assuming that the secret is already base64 decoded. if you managed to confirm that decode is always the case for AWS Go SDK, then please update accordingly to just return `secret.SecretBinary` directly.